[Hu Yuxin] iP

huyuxin0429 opened the pull request:
#DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ *SUPER FAST* to use

All you need to do is,
1. download it from [here](https://github.com/huyuxin0429/ip)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

 - [x] Managing tasks
 - [x] Managing deadlines
 - [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public static void main(String[] args) {
    Application.launch(Duke.class, args);
}
```

Commit:
Add Gradle support

huyuxin0429 committed:
finished level 1

huyuxin0429 committed:
added level 1

huyuxin0429 committed:
finished level 2

huyuxin0429 committed:
fixed list numbering bug

huyuxin0429 committed:
added Task class

huyuxin0429 committed:
added deadline, todo, events

huyuxin0429 committed:
added semi automatic testing

huyuxin0429 committed:
added some error handling

huyuxin0429 committed:
added deletion

huyuxin0429 committed:
refactored code to use enums

huyuxin0429 committed:
added formatting

huyuxin0429 committed:
added file save

huyuxin0429 committed:
Add data to .gitignore

huyuxin0429 committed:
Add date parsing

huyuxin0429 committed:
Merge branch 'branch-Level-7'

huyuxin0429 committed:
merged level-7 and level-8

huyuxin0429 committed:
Refactor code to become more OOP

huyuxin0429 committed:
made more OOP

huyuxin0429 committed:
Merge branch 'A-MoreOOP'

huyuxin0429 committed:
added package

huyuxin0429 committed:
Merge branch 'branch-A-Packages'

huyuxin0429 committed:
add tests to Event and Parser classes

huyuxin0429 committed:
Merge branch 'branch-A-JUnit'

huyuxin0429 committed:
add JAR file

huyuxin0429 committed:
fix event and deadline not responding bug

huyuxin0429 committed:
added javadocs

huyuxin0429 committed:
add find

huyuxin0429 committed:
Merge branch 'branch-Level-9'

huyuxin0429 committed:
add JavaDoc

zognin commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+
+    public Deadline(String description, String by) {
+        this(description, by, false);
+
```
Perhaps you may want to remove unnecessary line breaks at the end of this method to make it consistent with other methods? I noticed this in several other methods in other files too.

zognin commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
```
Would you be adding javadoc for your classes?

zognin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,137 @@
+
+
+
+public class Duke {
```
Would you be adding javadoc here too? And for other public methods?

zognin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,137 @@
+import java.io.IOException;
+import java.util.Scanner;
+
+
```
Similarly, perhaps you could remove unnecessary line breaks after imports, to make it consistent with other classes

zognin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,137 @@
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();
+
+        String[] commandAndParameter = Parser.inputParser(input);
```
```suggestion
        String[] commandAndParameter = Parser.parseInput(input);
```
Is there a reason why you deviated from the coding standard where 'Names representing methods must be verbs'?

zognin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,137 @@
+        String input = sc.nextLine();
+
+        String[] commandAndParameter = Parser.inputParser(input);
+        Command currentCommand = Parser.commandParser(commandAndParameter[0]);
```
```suggestion
        Command currentCommand = Parser.parseCommand(commandAndParameter[0]);
```

zognin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,137 @@
+                StringBuilder sb = new StringBuilder();
+                String[] descriptionAndTime;
+                switch (currentCommand) {
+                    case UNKNOWN:
```
Perhaps you may want to follow the java coding standard where there is no indentation for case clauses in a switch block

zognin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,137 @@
+            try {
+                System.out.println(currentCommand);
+                StringBuilder sb = new StringBuilder();
+                String[] descriptionAndTime;
```
```suggestion
                String[] descriptionAndTimeParts;
```
According to the java coding standard, it may be clearer if you use plural form to represent a collection of objects, like an array.

zognin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,137 @@
+        System.exit(0);
+    }
+
+
+
+
+
+
+
+
+
+
+
+
+
```
Similarly, it may be better to remove unnecessary new lines

zognin commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,49 @@
+
+public class Parser {
+    public static String[] inputParser(String input) {
+        String[] processed = input.split(" ", 2);
```
```suggestion
        String[] processedInputs = input.split(" ", 2);
```
Similarly, it may be more obvious that the variable represents a collection if it is named in plural form

zognin commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,49 @@
+        return Command.UNKNOWN;
+    }
+
+    public static String dateParser(String date) {
```
```suggestion
    public static String parseDate(String date) {
```
Similarly, you may want to name methods in verb form. I noticed some other occurrences of this too.

zognin commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,94 @@
+
+    public static void updateLocalFile(TaskList taskList) throws IOException {
+
+        ArrayList<Task> list = taskList.arrayList;
```
```suggestion
        ArrayList<Task> tasks = taskList.arrayList;
```
Perhaps you could name it in plural form? It may also be better to give it a name that informs other developers about what the variable contains

zognin commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,94 @@
+
+
+    public static ArrayList<Task> load(String path) {
+        ArrayList<Task> arrayList = new ArrayList<>(100);
```
Similar to the above, is there a reason why you chose to name it `arrayList`?

zognin commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+import java.util.ArrayList;
+
+public class TaskList {
+    protected ArrayList<Task> arrayList;
+    TaskList(ArrayList<Task> loaded) {
```
```suggestion
    protected ArrayList<Task> arrayList;
    
    TaskList(ArrayList<Task> loaded) {
```
You may want to add a line break to improve readability.

zognin commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,32 @@
+        }
+
+        this.at = date;
+    }
+    public Event(String description, String at) {
```
```suggestion
    }
    
    public Event(String description, String at) {
```
Perhaps you could add a line break to improve readability?

zognin commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+        StringBuilder sb = new StringBuilder();
+        sb.append("Got it. I've added this task: \n");
+        String parsedDate = Parser.dateParser(date).equals("") ? date : Parser.dateParser(date);
+//            if (Command.EVENT.equals(command)) {
+//                task = new Event(parameter, parsedDate);
+//            } else if (Command.TODO.equals(command)) {
+//                task = new Todo(parameter);
+//            } else {//if (Command.DEADLINE.equals(command)) {
+//                task = new Deadline(parameter, parsedDate);
+//            }
```
Is there a reason for keeping this commented out code?

huyuxin0429 committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

zognin commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+        return arrayList.size();
+    }
+
+    public void done(Integer i) {
```
```suggestion
    public void markAsDone(Integer i) {
```
Would it be better to give it an imperative verb name?

zognin commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+        Ui.printStatement(sb.toString());
+    }
+
+    public int size() {
```
```suggestion
    public int getSize() {
```

zognin commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,35 @@
+    public static void printStatement(String statement) {
+        System.out.println();
+        printLine();
+//        printPadding();
```
Similarly, is there a reason for keeping commented out code?

zognin commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,21 @@
+public class ParserTest {
+
+    @Test
+    public void emptyTextTest() {
```
```suggestion
    public void inputParser_emptyText_arrayOfEmptyStrings() {
```
Perhaps you could follow the java coding standard for naming tests? This applies to other test methods names too

huyuxin0429 committed:
added gradle

chunweii commented:
I think that you should consider using more packages to separate different components of your program. Currently, everything is just put under the package `duke`.

As mentioned by @zognin too, your code is lacking documentation. Consider adding javadocs to **all** public methods and classes.

chunweii commented on df sdfasdf:
```diff
@@ -0,0 +1,23 @@
+[33m90c4d2e[m[33m ([m[1;36mHEAD -> [m[1;32mmaster[m[33m, [m[1;33mtag: A-JavaDoc[m[33m, [m[1;31morigin/master[m[33m, [m[1;31morigin/HEAD[m[33m)[m master@{0}: merge branch-A-JavaDoc: Merge made by the 'recursive' strategy.
```
What is this file for? Please avoid committing such files to github. You can store them somewhere else or gitignore it.

huyuxin0429 committed:
Update src/main/java/duke/Duke.java

Changed inputParser to parseInput

Co-authored-by: Geng Ning <71375383+zognin@users.noreply.github.com>

huyuxin0429 committed:
Add assertions to Duke

The current code lacks assertions to ensure that
core task addition and deletion code is sound.

Add assertions to ensure that task increment/
decrement logic is working soundly.

huyuxin0429 committed:
Merge branch-A-Assertions to master

huyuxin0429 committed:
Add continuous integration to Duke

Current project does not have continuous integration enabled

This may cause desync of multiple people's work in the future

Adding CI integration in github helps solve this issue.

huyuxin0429 committed:
Merge branch 'branch-A-CI'

huyuxin0429 committed:
Amend CI file location

huyuxin0429 committed:
Add assertion for markAsDone

huyuxin0429 committed:
Merge pull request #1 from huyuxin0429/branch-A-Assertions

Add assertion for markAsDone

huyuxin0429 committed:
Revert "Add assertion for markAsDone"

huyuxin0429 committed:
Merge pull request #2 from huyuxin0429/revert-1-branch-A-Assertions

Revert "Add assertion for markAsDone"

huyuxin0429 committed:
Add assertion for deletion in TaskList

huyuxin0429 committed:
Merge pull request #3 from huyuxin0429/branch-A-Assertions

Add assertion for deletion in TaskList

huyuxin0429 committed:
Merge branch 'master' of https://github.com/huyuxin0429/ip

huyuxin0429 committed:
Added comments for run

huyuxin0429 committed:
Merged with branch-A-CodingStandard

huyuxin0429 committed:
add README user guide

huyuxin0429 committed:
rename UI to Ui

huyuxin0429 committed:
change Ui.png to a better picture

huyuxin0429 committed:
refactor javafx UI portion to level-4

huyuxin0429 committed:
refactor javafx UI portion to level-4

huyuxin0429 committed:
merge javafx refactor to master

huyuxin0429 committed:
changed visuals for ui, made size not changable
